src/main/java/database/managers/PlayerCreatorManager.java

```diff
@@ -0,0 +1,110 @@
+package database.managers;
+
+import interfaces.State;
+import playercreation.PlayerCreatorInteractor;
+import playercreation.PlayerQuestion;
+import playercreation.PlayerQuestionState;
+import playercreation.PlayerConfirmState;
+import core.StateManager;
+
+import java.util.ArrayList;
+
+public class PlayerCreatorManager extends StateManager {
+    /** A controller class that helps create a new Player character at the beginning of the game.
+     * Attributes:
+     * currPlayerQuestion: The current question about the Player character being asked of the user. An enum of
+     *                     type PlayerQuestion.
+     * completedQuestions: A List of completed PlayerQuestions that have already been filled by the user. Does not
+     *                     include CONFIRM or COMPLETE.
+     * creatorInteractor: A PlayerCreatorInteractor that stores information about the Player until all questions have
+     *                    been answered by the user.
+     */
+
+    private PlayerQuestion currPlayerQuestion;
+    private final ArrayList<PlayerQuestion> completedQuestions;
+    private final PlayerCreatorInteractor creatorInteractor;
+
+    public PlayerCreatorManager() {
+        // Initializes a new PlayerCreatorManager.
+        this.completedQuestions = new ArrayList<>();
+        this.creatorInteractor = new PlayerCreatorInteractor();
+        initialize();
+    }
+
+    @Override
+    public void initialize() {
+        // Initializes the PlayerCreatorManager.
+        currPlayerQuestion = PlayerQuestion.NAME;
+        currState = new PlayerQuestionState(currPlayerQuestion, this.creatorInteractor);
+    }
+
+    public PlayerQuestion getCurrPlayerQuestion() {
+        // Return the current PlayerQuestion.
+        return this.currPlayerQuestion;
+    }
+
+    public State getCurrState() {
+        // Return the current State.
+        return this.currState;
+    }
+
+    public ArrayList<PlayerQuestion> getCompletedQuestions() {
+        // Return the completedQuestions.
+        return this.completedQuestions;
+    }
+
+    @Override
+    public State nextState(String input) {
+        // Switches the question state to the next state in order to ask for more input from the user.
+        if (currPlayerQuestion == PlayerQuestion.NAME ||
+                currPlayerQuestion == PlayerQuestion.DESCRIPTION ||
+                currPlayerQuestion == PlayerQuestion.SKILLTYPE) {
+            if (currState.getInputValidator().parseAndValidate(input) != null) {
```

Comment:
this would already be executed in inputHandler no need to do it again, you are guaranteed a valid input if getInputValidator does not return null

This is in a lot of files so I wont write a comment for each one